src/__tests__/emailverification.js

```diff
@@ -0,0 +1,149 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import app from '../index';
+
+import emailverification from '../controllers/emailController';
+import { emailVerifyToken } from '../utils/index';
+import { emailRegex } from '../validation/emailValidation';
+import { idUnset, idWrong } from '../__mocks__/emailVerification';
+
+chai.use(chaiHttp);
+chai.use(sinonChai);
+
+const { expect } = chai;
+
+let request;
+let tokenEmail;
+
+describe('EMAIL ROUTE', () => {
+  before(async () => {
+    request = chai.request(app).keepOpen();
+  });
+
+  afterEach(() => sinon.restore());
+
+  describe('EMAIL VERIFICATION ROUTE', () => {
+    it('should have a status of 200 when message is sent succesfully', async () => {
+      const body = {
+        email: 'akp.ani@yahoo.com',
+        firstName: 'Aniefiok',
+        lastName: 'Akpan'
+      };
+      const response = await request.post('/api/v1/users/email/test').send(body);
+      tokenEmail = response.body.data.token;
+      expect(response.body.status).to.equal(200);
+      expect(response.body).to.be.a('object');
+    }).timeout(0);
+
+    it('should have a status of 403 email is not present before sending', async () => {
+      const response = await request.post('/api/v1/users/email/test');
+      expect(response.body.status).to.equal(403);
+      expect(response.body).to.be.a('object');
+    }).timeout(0);
+
+    it('should have a status of 403 when BODY is not present', async () => {
+      const response = await request.post('/api/v1/users/email/test');
+      expect(response.body.status).to.equal(403);
+      expect(response.body).to.be.a('object');
+    }).timeout(0);
+
+    it('should have a status of 403 and a messsage of "please fill in the required inputs"  when some body is not present', async () => {
```

Comment:
the message `please fill in the required inputs` is too vague. you should try and be more specific i.e `email, firstname is/are required`. This will help users solve the issue faster.

Reply:
Done